source/extensions/filters/http/aws_lambda/config.cc

```diff
@@ -0,0 +1,64 @@
+#include "extensions/filters/http/aws_lambda/config.h"
+
+#include "envoy/extensions/filters/http/aws_lambda/v3/aws_lambda.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "common/common/fmt.h"
+
+#include "extensions/common/aws/credentials_provider_impl.h"
+#include "extensions/common/aws/signer_impl.h"
+#include "extensions/common/aws/utility.h"
+#include "extensions/filters/http/aws_lambda/aws_lambda_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AwsLambdaFilter {
+constexpr auto service_name = "lambda";
+namespace {
+std::string extractRegionFromArn(absl::string_view arn) {
+  auto parsed_arn = parse_arn(arn);
+  if (parsed_arn.has_value()) {
+    return parsed_arn->region();
+  }
+  throw EnvoyException(fmt::format("Invalid ARN: {}", arn));
+}
+} // namespace
+
+Http::FilterFactoryCb AwsLambdaFilterFactory::createFilterFactoryFromProtoTyped(
+    const envoy::extensions::filters::http::aws_lambda::v3::Config& proto_config,
+    const std::string& stats_prefix, Server::Configuration::FactoryContext& context) {
+
+  (void)stats_prefix;
+  auto credentials_provider =
+      std::make_shared<Extensions::Common::Aws::DefaultCredentialsProviderChain>(
+          context.api(), Extensions::Common::Aws::Utility::metadataFetcher);
+
+  const std::string region = extractRegionFromArn(proto_config.arn());
```

Comment:
could this be a string view?  I'm pretty sure the lifetime of the arn won't be an issue for signing the request.